src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -102,9 +102,14 @@ public static function createConnection(string $dsn, array $options = [])
             throw new CacheException(sprintf('Cannot find the "redis" extension nor the "predis/predis" package: "%s".', $dsn));
         }
 
-        $params = preg_replace_callback('#^'.$scheme.':(//)?(?:(?:[^:@]*+:)?([^@]*+)@)?#', function ($m) use (&$auth) {
-            if (isset($m[2])) {
-                $auth = $m[2];
+        $params = preg_replace_callback('#^'.$scheme.':(//)?(?:(?:(?<user>[^:@]*+):)?(?<password>[^@]*+)@)?#', function ($m) use (&$auth) {
+            if (isset($m['password'])) {
+                // if user is not `default` then it is acl user
```

Comment:
where does "default" come from? is that a new convention you're adding?
right now, the convention is to ignore the user part, so that we could consider that the empty string is the no-ACL case.
Wouldn't that be more appropriate?

Reply:
You are right. I will REMOVE that check  to keep clean convention:
has useryname (://user:pass@) - ACL
no user (://pass@) - no ACL default user